python/shotgun_data/sgdata.py

```diff
@@ -532,6 +535,23 @@ def _get_thumbnail_path(url, bundle):
             
         return path_to_cached_thumb
 
+    def __ensure_sg_connection_initialized(self):
+        """
+        Populates self.__sg unless it has already been set.
+        
+        This allows for a workflow when the sg connection is 
+        created just in time before a sg operation, allowing
+        maximum speed for cache I/O tasks running in the data
+        fetcher (like grabbing cached thumbnails from disk). 
+        """
+        if self.__sg is None:
```

Comment:
One safer way to have this functionality would probably be to have a property _sg that lazily initializes __sg and use self.sg in the code?